video/packet/encode.go

```diff
@@ -0,0 +1,209 @@
+package packet
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const AppDirName = "gioplayer"
+const VideosDirName = "videos"
+const CachedFileName = "filepath"
+const AudioFileName = "audio.aac"
+
+func (p *Controller) FilePath() string {
+	return p.filePath
+}
+
+func (p *Controller) VideoDecodingState() *CommonState {
+	return &p.decodingVideoState
+}
+
+func (p *Controller) AudioDecodingState() *CommonState {
+	return &p.decodingAudioState
+}
+
+func (p *Controller) EncodingState() *CommonState {
+	return &p.encodingState
+}
+
+func (p *Controller) StartTime() int64 {
+	return p.startTime
+}
+
+func (p *Controller) EndTime() int64 {
+	p.endTimeMutex.RLock()
+	defer p.endTimeMutex.RUnlock()
+	return p.endTime
+}
+
+func (p *Controller) SetStartTime(s int64) {
+	p.startTime = s
+}
+
+func (p *Controller) SetEndTime(s int64) {
+	p.endTimeMutex.Lock()
+	defer p.endTimeMutex.Unlock()
+	p.endTime = s
+}
+
+func (p *Controller) encode() {
+	var status Status
+	var message string
+	defer func() {
+		if r := recover(); r != nil {
+			status = Error
+			message = fmt.Sprintf("%s", r)
+			p.encodingState.setState(status, message)
+		}
+		if status == Running {
+			status = Error
+			message = "Unknown Error"
+		}
+		p.encodingState.setState(status, message)
+	}()
+
+	status = Running
+	message = "Encoding started..."
+	p.encodingState.setState(status, message)
+
+	dir := os.TempDir()
+	dir = filepath.Join(dir, AppDirName, VideosDirName)
+
+	cachedPath := filepath.Join(dir, CachedFileName)
+	audioPath := filepath.Join(dir, AudioFileName)
+	p.audioPath = audioPath
+
+	_, err := os.Stat(cachedPath)
+	fileExist := !errors.Is(err, os.ErrNotExist)
+
+	if fileExist {
+		file, err := os.Open(cachedPath)
+		if err == nil {
+			b, err := ioutil.ReadAll(file)
+			if err == nil {
+				previousFilePath := string(b)
+				if previousFilePath != p.FilePath() {
+					fileExist = false
+				}
+			}
+		}
+		err = file.Close()
+		if err != nil {
+			fileExist = false
+		} else {
+			message = fmt.Sprintf("File exist, using cached path %s", cachedPath)
+			p.encodingState.setState(status, message)
+		}
+	}
+
+	if !fileExist {
+		err = os.RemoveAll(dir)
+		if err != nil {
+			message = err.Error()
+			status = Error
+			return
+		}
+		err = os.MkdirAll(dir, os.ModePerm)
+		if err != nil {
+			message = err.Error()
+			status = Error
+			return
+		}
+	}
+
+	argSlice := []string{"-i", p.FilePath(), "-show_entries", "format=duration", "-v", "quiet", "-of", "csv=p=0"}
+	command := exec.Command("ffprobe", argSlice...)
+	b, err := command.CombinedOutput()
+	if err != nil {
+		message = err.Error()
+		status = Error
+		return
+	}
+
+	endTimeStr := strings.Split(string(b), "\n")[0]
+	if runtime.GOOS == "windows" {
+		endTimeStr = strings.Split(string(b), "\r\n")[0]
+	}
```

Comment:
```suggestion
	endTimeStr := strings.Split(string(b), filepath.PathSeparator)[0]
```
I think this is a little cleaner for handling the windows case. However, I have another recommendation that may eliminate this entirely.